addons/mail/sfu_server/src/models/channel.js

```diff
@@ -0,0 +1,224 @@
+    /** @type {Map<string, Channel>} */
+    static recordsByIssuer = new Map();
+
+    /** @type {string} */
+    remoteAddress;
+    /** @type {string} */
+    uuid;
+    /** @type {string} short of uuid for logging */
+    name;
+    /** @type {import("mediasoup").types.Router}*/
+    router;
+    /** @type {Map<number, Session>} */
+    sessions = new Map();
+    /** @type {import("mediasoup").types.Worker}*/
+    _worker;
+    /** @type {NodeJS.Timeout} */
+    _closeTimeout;
+
+    /**
+     * @returns {Channel}
+     * @param {string} remoteAddress
+     * @param {string} issuer
+     */
+    static async create(remoteAddress, issuer) {
+        const safeIssuer = remoteAddress + issuer;
+        const oldChannel = Channel.recordsByIssuer.get(safeIssuer);
+        if (oldChannel) {
+            logger.info(`reusing channel ${oldChannel.uuid}`);
+            return oldChannel;
+        }
+        const { worker, releaseWorker } = getWorker();
+        const router = await worker.createRouter({
+            mediaCodecs,
+        });
+        const channel = new Channel(remoteAddress, { worker, router });
+        Channel.recordsByIssuer.set(safeIssuer, channel);
+        logger.info(`created channel ${channel.uuid} for ${remoteAddress}`);
+        channel.once("close", () => {
+            releaseWorker();
+            Channel.recordsByIssuer.delete(safeIssuer);
+        });
+        channel.setCloseTimeout(true);
+        return channel;
+    }
+
+    /**
+     * @param {string} uuid
+     * @param {number} sessionId
+     */
+    static join(uuid, sessionId) {
+        const channel = Channel.records.get(uuid);
+        if (!channel) {
+            throw new Error(`channel [${uuid}] does not exist`);
+        }
+        const session = channel.join(sessionId);
+        return { channel, session };
+    }
+
+    static closeAll() {
+        for (const channel of Channel.records.values()) {
+            channel.close();
+        }
+        Channel.records.clear();
+    }
+
+    /**
+     * @returns {{ channelCount: number, sessionCount: number}}
+     */
+    static getStats() {
+        return {
+            channelCount: Channel.records.size,
+            sessionCount: [...Channel.records.values()].reduce(
+                (acc, channel) => acc + channel.sessions.size,
+                0
+            ),
+        };
+    }
+
+    /**
+     * @param {string} remoteAddress
+     * @param {Object} param1
+     * @param {import("mediasoup").types.Worker} param1.worker
+     * @param {import("mediasoup").types.Router} param1.router
+     */
+    constructor(remoteAddress, { worker, router }) {
+        super();
+        this.remoteAddress = remoteAddress;
+        this.uuid = crypto.randomUUID();
+        this.name = `${remoteAddress}*${this.uuid.slice(-5)}`;
+        this.router = router;
+        this._worker = worker;
+        this._onSessionClose = this._onSessionClose.bind(this);
+        Channel.records.set(this.uuid, this);
```

Comment:
if we keep both records and recordsByIssuer they should probably be set and unset in the same way